dot/sync/chain_sync.go

```diff
@@ -691,17 +712,43 @@ func handleReadyBlock(bd *types.BlockData, pendingBlocks DisjointBlockSet, ready
 }
 
 // determineSyncPeers returns a list of peers that likely have the blocks in the given block request.
-func (cs *chainSync) determineSyncPeers(_ *network.BlockRequestMessage) []peer.ID {
+func (cs *chainSync) determineSyncPeers(req *network.BlockRequestMessage, peersTried map[peer.ID]struct{}) []peer.ID {
+	var start uint64
+	if req.StartingBlock.IsUint64() {
+		start = req.StartingBlock.Uint64()
+	}
+
 	cs.RLock()
 	defer cs.RUnlock()
 
+	// if we're currently ignoring all our peers, clear out the list.
+	if len(cs.peerState) == len(cs.ignorePeers) {
```

Comment:
Is this so we don't find the node in a state where its stuck not listening to any peers?

Reply:
yeah, not "not listening" exactly but basically ignoring every peer for syncing purposes ;p